include/zephyr/drivers/can.h

```diff
@@ -172,8 +172,12 @@ struct can_frame {
 	 *
 	 * @note @kconfig{CONFIG_CAN_RX_TIMESTAMP} must be selected for this
 	 * field to be available.
+	 *
+	 * The full width of the timestamp field is not used by all CAN drivers.
+	 * Most of them use 16-bit of the timestamp and only CANXL NXP S32 driver uses
+	 * full width of the timestamp.
 	 */
-	uint16_t timestamp;
+	uint32_t timestamp;
```

Comment:
we can support 32bits timestamp? you suggested before https://github.com/zephyrproject-rtos/zephyr/pull/52450#discussion_r1178306712